File: 07a/python/main.py
with open('input.txt', 'rt') as f:
    crabs = [int(x) for x in f.readline().split(',')]

costs = [sum(abs(c - p) for c in crabs) for p in range(max(crabs))]
best_fuel = min(costs)
best_pos = costs.index(best_fuel)

print(best_fuel, best_pos)

File: 07a/ruby/main.rb
FILENAME = "input.txt"
f = File.open(FILENAME)
pos = f.read.split(",").map(&:to_i)

# This is lazy but it'll work...
# Figure out how much fuel it'll take to get to each position
min = 0
max = pos.max + 1
options = (min..max).map do |option|
  pos.map{|p| p-option}.map(&:abs).sum
end

p options.min
